qcodes/instrument_drivers/Galil/dmc_41x3.py

```diff
@@ -0,0 +1,964 @@
+from qcodes.instrument.channel import InstrumentChannel
+from qcodes.utils.validators import Enum, Ints, Multiples
+
+try:
+    import gclib
+except ImportError as e:
+    raise ImportError(
+        "Cannot find gclib library. Download gclib installer from "
+        "https://www.galil.com/sw/pub/all/rn/gclib.html and install Galil "
+        "motion controller software for your OS. Afterwards go "
+        "to https://www.galil.com/sw/pub/all/doc/gclib/html/python.html and "
+        "follow instruction to be able to import gclib package in your "
+        "environment."
+    ) from e
+
+
+class GalilMotionController(Instrument):
+    """
+    Base class for Galil Motion Controller drivers
+    """
+
+    def __init__(self, name: str, address: str, **kwargs: Any) -> None:
+        """
+        Initializes and opens the connection to the Galil Motion Controller
+
+        Args:
+            name: name for the instrument
+            address: address of the controller
+        """
+        super().__init__(name=name, **kwargs)
+        self.g = gclib.py()
+        self.address = address
+        self.open()
+
+    def open(self) -> None:
+        """
+        Open connection to Galil motion controller. This method assumes that
+        the initial mapping of Galil motion controller's hardware's mapping
+        to an IP address is done using GDK and the IP address in burned in.
+        This applies that Motion controller no more requests for an IP address
+        and a connection to the Motion controller can be done by the IP
+        address burned in.
+        """
+        self.g.GOpen(self.address + " --direct -s ALL")
+
+    def get_idn(self) -> Dict[str, Optional[str]]:
+        """
+        Get Galil motion controller hardware information
+        """
+        data = self.g.GInfo().split(" ")
+        idparts: List[Optional[str]] = [
+            "Galil Motion Control, Inc.",
+            data[1],
+            data[4],
+            data[3][:-1],
+        ]
+
+        return dict(zip(("vendor", "model", "serial", "firmware"), idparts))
+
+    def write_raw(self, cmd: str) -> None:
+        """
+        Write for Galil motion controller
+        """
+        self.g.GCommand(cmd + "\r")
+
+    def ask_raw(self, cmd: str) -> str:
+        """
+        Asks/Reads data from Galil motion controller
+        """
+        return self.g.GCommand(cmd + "\r")
+
+    def timeout(self, val: int) -> None:
+        """
+        Sets timeout for the instrument
+
+        Args:
+            val: time in milliseconds. -1 disables the timeout
+        """
+        if val < -1:
+            raise RuntimeError("Timeout value cannot be set less than -1")
+
+        self.g.GTimeout(val)
+
+    def close(self) -> None:
+        """
+        Close connection to the instrument
+        """
+        self.g.GClose()
+
+    def motion_complete(self, axes: str) -> None:
+        """
+        Wait for motion to complete for given axes
```

Comment:
is my understanding correct?
```suggestion
        Wait for motion to complete for given axes, e.g. ``"A"``
        for axis A, ``"ABC"`` for axes A, B, and C
```

Reply:
yes